### PR TITLE
jwt: use pre-computed JWT header

### PR DESCRIPTION
**Pull request by amarkevich**
JWT header is unmodifable and can be used as constant.

Signed-off-by: Alexey Markevich <buhhunyx@gmail.com>

### CODE REVIEW

File: subsys/jwt/jwt.c [hunk 1]
```diff
@@ -127,15 +127,16 @@ static int base64_append_bytes(const char *bytes, size_t len,
...
-	char *typ;
-	char *alg;
-};
-
-static struct json_obj_descr jwt_header_desc[] = {
-	JSON_OBJ_DESCR_PRIM(struct jwt_header, alg, JSON_TOK_STRING),
-	JSON_OBJ_DESCR_PRIM(struct jwt_header, typ, JSON_TOK_STRING),
-};
+static const char jwt_header[] =
+#ifdef CONFIG_JWT_SIGN_RSA
```
Comment (galak) [inline subsys/jwt/jwt.c:131]: Can you add a bit more to the comments about exactly how these values are created.

**Comment by amarkevich** [comment]
@galak added comment to mention base64UrlEncode

**Comment by amarkevich** [comment]
This improvement require less memory:
before: sizeof(jwt_header_desc) + sizeof("JWT") + sizeof("ES256") = 32 + 4 + 6 = 42
after: sizeof(jwt_header) = 37
Also new code executed a little bit faster; using nrf9160dk got ~1ms

**Comment by galak** [comment]
> @galak added comment to mention base64UrlEncode

I guess I was looking for something a bit more detailed that would explain how someone else could generate/reproduce that same encoded value.

**Comment by amarkevich** [comment]
I used https://jwt.io/
Put required header into corresponding editbox and have ready to use value from Encoded area (red part, up to first dot)

**Pull request closed without merging**